File: typescript/submit-negotiation.ts
import { createNexezClient } from '@nexez/agent-sdk'

const nexez = createNexezClient({ buyerAgent: 'nexez-typescript-example' })

const proposal = {
  slug: 'nexez-agent-negotiation-lab',
  offer: 'services-0',
  query: 'Buyer wants a one-week agent negotiation sprint.',
  budget: 'USD 2100',
  timeline: 'next week',
  requestedTerms: {
    scope: 'Discovery call, agent-readable offer review, and dry-run guidance.',
  },
}

const dryRun = await nexez.validateNegotiation(proposal)
console.log({ dryRun })

// In a real buyer agent, stop here and ask the buyer:
// "Approve sending this proposal to the seller?"
const approvedByBuyer = false
if (!approvedByBuyer) {
  console.log('Buyer approval required before submitNegotiation.')
  process.exit(0)
}

const submitted = await nexez.submitNegotiation({
  ...proposal,
  contact: 'buyer@example.com',
  approvalToken: dryRun.approvalToken,
  userApproved: true,
}, {
  idempotencyKey: crypto.randomUUID(),
})
console.log({
  submitted: {
    ok: submitted.ok,
    status: submitted.status,
    negotiationId: submitted.negotiationId,
    decisionPending: submitted.decisionPending,
  },
})

if (submitted.negotiationId && submitted.statusToken) {
  const status = await nexez.waitForNegotiationDecision({
    negotiationId: submitted.negotiationId,
    statusToken: submitted.statusToken,
    timeoutMs: 30_000,
    intervalMs: 2_000,
  })
  console.log({ status })
}

File: python/submit_negotiation.py
from nexez_agent_sdk import NexezApiError, create_client
from uuid import uuid4


def main() -> None:
    nexez = create_client(buyer_agent="nexez-python-example")

    slug = "nexez-agent-negotiation-lab"
    offer = "services-0"
    proposal = {
        "slug": slug,
        "offer": offer,
        "query": "Buyer wants a one-week agent negotiation sprint.",
        "budget": "USD 2100",
        "timeline": "next week",
        "requested_terms": {
            "scope": "Discovery call, agent-readable offer review, and dry-run guidance.",
        },
    }

    dry_run = nexez.validate_negotiation(proposal)
    print({"dry_run": dry_run})

    # In a real buyer agent, stop here and ask the buyer:
    # "Approve sending this proposal to the seller?"
    approved_by_buyer = False
    if not approved_by_buyer:
        print("Buyer approval required before submit_negotiation.")
        return

    submitted = nexez.submit_negotiation(
        proposal,
        contact="buyer@example.com",
        approval_token=dry_run.get("approvalToken"),
        user_approved=True,
        idempotency_key=uuid4().hex,
    )
    print(
        {
            "submitted": {
                "ok": submitted.get("ok"),
                "status": submitted.get("status"),
                "negotiationId": submitted.get("negotiationId"),
                "decisionPending": submitted.get("decisionPending"),
            }
        }
    )

    negotiation_id = submitted.get("negotiationId")
    status_token = submitted.get("statusToken")
    if negotiation_id and status_token:
        try:
            status = nexez.wait_for_negotiation_decision(
                negotiation_id,
                status_token,
                timeout=30.0,
                poll_interval=2.0,
            )
            print({"status": status})
        except TimeoutError:
            print({"status": "timed_out", "next": "Check this negotiation again later."})


if __name__ == "__main__":
    try:
        main()
    except NexezApiError as error:
        print({"error": str(error), "status": error.status, "body": error.body})
